### PR TITLE
Sonar issues

### PR DESCRIPTION
**Pull request by mookkiah**
"throws" declarations should not be superfluous (java:S1130)
Local variable and method parameter names should comply with a naming convention (java:S117)
Synchronized classes Vector, Hashtable, Stack and StringBuffer should not be used (java:S1149)
Methods should not return constants (java:S3400)
Remove the declaration of thrown exception 'org.apache.hop.core.exception.HopException', as it cannot be thrown from method's body.
Collection.isEmpty() should be used to test for emptiness (java:S1155)
Local variables should not shadow class fields (java:S1117)
Unused assignments should be removed (java:S1854)

### CODE REVIEW

File: plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapConnection.java [hunk 1]
```diff
@@ -249,21 +249,22 @@ public void search( String searchBase, String filter, int limitRows, String[] at
...
         Control[] ctls = new Control[ nrCtl ];
         int index = 0;
         if ( ctlk != null ) {
-          ctls[ index++ ] = ctlk;
+          ctls[ index ] = ctlk;
+          index++;
         }
         if ( ctlp != null ) {
-          ctls[ index++ ] = ctlp;
+          ctls[ index ] = ctlp;
```
Comment (hansva) [inline plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapConnection.java:267]: Missing index++

**Review by hansva — CHANGES_REQUESTED** [review]

File: plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapConnection.java (hunk 1 above)
Comment (mookkiah) [reply plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapConnection.java:267]: It was unnecessary increment.
https://sonarcloud.io/project/issues?id=hop&issues=AXG4QflqIq6iCIhl5L7h&open=AXG4QflqIq6iCIhl5L7h

File: plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapProtocol.java [hunk 2]
```diff
@@ -63,103 +67,109 @@ public InitialLdapContext getCtx() {
...
   /**
-   * Method signature used by factory to get display name, method should exist in every ldap protocol
+   * Method signature used by factory to get display name, method should exist in every ldap
+   * protocol
    *
    * @return the display name
    */
   public static String getName() {
-    return "LDAP";
+    return NAME;
```
Comment (mookkiah) [inline plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapProtocol.java:97]: Bug - Returning empty string. Expected "LDAP"

File: plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapConnection.java [hunk 3]
```diff
@@ -47,15 +48,14 @@
...
 import javax.naming.ldap.SortControl;
-import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
 public class LdapConnection {
-  private static Class<?> PKG = LdapInputMeta.class; // for i18n purposes, needed by Translator!!
```
Comment (mookkiah) [inline plugins/transforms/ldap/src/main/java/org/apache/hop/pipeline/transforms/ldapinput/LdapConnection.java:58]: making it as final is good idea.

**Pull request merged**